lesson_13/mihai_farcas/homework_polygons.py

```diff
@@ -0,0 +1,63 @@
+from math import sqrt
+
+
+class PerimetruMixin:
+
+    def __init__(self):
+        self.sides = None
+
+    def perimetru(self):
+        class_name = type(self).__name__
+        return 'The perimeter for {} is : {}'.format(class_name, sum(self.sides))
+
+
+class Polygons(PerimetruMixin):
+
+    def __init__(self, *args):
+        self.sides = args
+
+    def __str__(self):
+        no_of_sides = len(self.sides)
+        return '{} is the number of sides'.format(no_of_sides)
+
+    def display(self):
+        for side_index, length in enumerate(self.sides, start=1):
+            print('Side {} with length: {}'.format(side_index, length))
+
+
+class Triangle(Polygons):
+    # `s(s-a)(s-b)(s-c) ** 0.5` where s = `(a+b+c) / 2`
+    def __init__(self, *args):
+        super().__init__(*args)
+
+    def area(self):
+        s1, s2, s3 = self.sides
+        s_p = sum(self.sides) / 2
+        return (s_p * (s_p - s1) * (s_p - s2) * (s_p - s3)) ** 0.5
+
+
+class Square(Polygons):
+
+    def __init__(self, *args):
+        super().__init__(*args)
+
+    def area(self):
+        side, *_ = self.sides
+        return side ** 2
+
+    def from_area(area):
```

Comment:
This method is supposed to return a new Square object. The logic of getting the sides is correct, but with the sides, we need to create a new object. In order to create the new object we need the class object (cls).

What we can use from python to achieve this?